amp-pwa/src/components/amp-document.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+
+/**
+ * Fetches the AMP document at a given `src` URL and renders it via Shadow DOM.
+ */
+export default class AMPDocument extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // 'offline' is set to true if and when the document fetch fails.
+    this.state = {'offline': false};
+
+    /**
+     * `window.AMP` is set by the AMP runtime when it finishes loading.
+     * @private
+     */
+    this.ampReadyPromise_ = new Promise(resolve => {
+      (window.AMP = window.AMP || []).push(resolve);
+    });
+
+    /**
+     * Child element that the AMP document will be added as a shadow root to.
+     * @private
+     */
+    this.container_ = null;
+
+    /**
+     * XMLHTTPRequest that fetches the AMP document.
+     * @private
+     */
+    this.xhr_ = null;
+  }
+
+  componentDidMount() {
+    this.fetchAndAttachAMPDoc_(this.props.src);
```

Comment:
Generally, we do full camel-case style guide. E.g. `fetchAndAttachAmpDoc_`. If it doesn't conflict with React - let's switch.